features/375-static-cdn/cdn.md

```diff
@@ -0,0 +1,64 @@
+# Static Asset CDN
+
+## Architecture
+
+### Network Diagram
+
+[![](https://mermaid.ink/img/eyJjb2RlIjoiZ3JhcGggVERcbiAgSU5FVFtDbGllbnRzXSAtLT58Q3VzdG9tIERvbWFpbnwgQ0ROXG4gIElORVQgLS0-IHxEZWZhdWx0IERvbWFpbnwgSUdcbiAgQ0ROW0NETiBOZXR3b3JrXSAtLT4gSUdcbiAgSUdbSW5ncmVzc10gLS0-IEdXe0dhdGV3YXl9XG4gIEdXIC0tPnxzdGF0aWN8IE9bT2JqZWN0IFN0b3JhZ2VdXG4gIEdXIC0tPnxodHRwLXNlcnZpY2V8IE1bTWljcm8tc2VydmljZXNdXG5cdFx0IiwibWVybWFpZCI6eyJ0aGVtZSI6ImRlZmF1bHQifX0)](https://mermaid-js.github.io/mermaid-live-editor/#/edit/eyJjb2RlIjoiZ3JhcGggVERcbiAgSU5FVFtDbGllbnRzXSAtLT58Q3VzdG9tIERvbWFpbnwgQ0ROXG4gIElORVQgLS0-IHxEZWZhdWx0IERvbWFpbnwgSUdcbiAgQ0ROW0NETiBOZXR3b3JrXSAtLT4gSUdcbiAgSUdbSW5ncmVzc10gLS0-IEdXe0dhdGV3YXl9XG4gIEdXIC0tPnxzdGF0aWN8IE9bT2JqZWN0IFN0b3JhZ2VdXG4gIEdXIC0tPnxodHRwLXNlcnZpY2V8IE1bTWljcm8tc2VydmljZXNdXG5cdFx0IiwibWVybWFpZCI6eyJ0aGVtZSI6ImRlZmF1bHQifX0)
+
+### CDN-enabled Custom Domain Flow
+
+[Graph](https://mermaid-js.github.io/mermaid-live-editor/#/edit/eyJjb2RlIjoic2VxdWVuY2VEaWFncmFtXG4gICAgcGFydGljaXBhbnQgREVWIGFzIERldmVsb3BlclxuICAgIHBhcnRpY2lwYW50IEROUyBhcyBETlNcbiAgICBwYXJ0aWNpcGFudCBDVFJMIGFzIENvbnRyb2xsZXJcbiAgICBwYXJ0aWNpcGFudCBDRE4gYXMgQ0ROIFByb3ZpZGVyXG4gICAgcGFydGljaXBhbnQgSzhTIGFzIEt1YmVybmV0ZXNcbiAgICBERVYtPj4rQ1RSTDogQWRkIGN1c3RvbSBkb21haW5cbiAgICBDVFJMLT4-K0NETjogUHJvdmlzaW9uIENETiByZXNvdXJjZXNcbiAgICBDRE4tPj4tQ1RSTDogQ05BTUUgcmVjb3JkXG4gICAgQ1RSTC0-Pi1ERVY6IFZlcmlmaWNhdGlvbiBjaGFsbGVuZ2UgKEEvQ05BTUUgJiBUWFQgcmVjb3JkKVxuICAgIERFVi0-PkROUzogVXBkYXRlIEROUyBSZWNvcmRcbiAgICBERVYtPj4rQ1RSTDogVHJpZ2dlciB2ZXJpZmljYXRpb25cbiAgICBDVFJMLT4-RE5TOiBWZXJpZnkgRE5TIHJlY29yZHNcbiAgICBDVFJMLT4-SzhTOiBDcmVhdGUgSW5ncmVzcyAmIFByb3Zpc2lvbiBUTFMgQ2VydFxuICAgIEs4Uy0-PkNUUkw6IFRMUyBDZXJ0XG4gICAgQ1RSTC0-PkNETjogU3luY2hyb25pemUgVExTIENlcnRcbiAgICBDVFJMLT4-LURFVjogQ3VzdG9tIGRvbWFpbiBpcyByZWFkeSIsIm1lcm1haWQiOnsidGhlbWUiOiJkZWZhdWx0In19)
```

Comment:
The sequence diagram is great! But maybe we will want to add one more section about the implementation details that how we synchronize the TLS cert to the cdn provider? We will do it in the current controller or another k8s crd controller?

Reply:
I was leaving it intentionally vague, since it is an implementation details that would not have visible external effect. In the diagram, 'Controller' entity represents the logical cluster controller, i.e. existing controller and k8s CRD controller (if we opt to use this approach).